### ISSUE TITLE
Does not detect hyperlink support in other terminals

### ISSUE THREAD

**Issue by nogweii** [piotrmurach/tty-link#1]
### Describe the problem

Alacritty, Kitty, and KDE's Konsole all support the hyperlink escape code. It's not just iTerm & VTE based terminals. 

### Steps to reproduce the problem

```
puts TTY::Link("foo", "https://google.com")
```

### Actual behaviour

To print out the hyperlink.

### Expected behaviour

Fallback output was printed instead.

### Describe your environment

* OS version: Arch Linux
* Ruby version: 3.2.2
* TTY::Link version: 0.1.1

**Comment by piotrmurach** [comment]
Thanks for submitting this issue. 🙏 

>  It's not just iTerm & VTE based terminals.

At the time this gem was created this was the case. These were the first terminals to support hyperlinks. The world has changed since. Alacritty has added support only recently(around 2022).

Therefore, I disagree that this is a 'bug'. More of a feature request. I'm sure you can appreciate that there are more terminals adding hyperlinks support and more terminals appearing all the time. This is in a way an 'impossible' task to create detection that works for every terminal including the ones that don't even exist yet. However, we can certainly try to support the most common ones.

I'd welcome help here. It can be in many forms. Either a PR that implements support for a given terminal. Alternatively, information on how to detect the terminal type from ENV and whether there is a version requirement. For example, use this name from the `TERM` variable and apply it to any version higher than 3.0 using `X_VERSION` variable. Ideally, this would be tested by installing a given terminal and verifying that detection works.

Further, we could add a configuration option to the `link_to` to allow overriding automatic detection. For example, it could take one of `always|auto|never`:

```ruby
TTY::Link.link_to("ttytoolkit.org", "https://ttytoolkit.org", hyperlink: :always)
```

Similarly, provide an environment variable to allow configuration for a tool that uses `tty-link`:

```shell
TTY_LINK_HYPERLINK=never
```

**Comment by nogweii** [comment]
The official mechanism would be to query the `terminfo` database for the capability, using something like [this library](https://github.com/genki/ruby-terminfo). I'm not certain what the hyperlink capability is called.

In the mean time, adding an parameter would be a good stop-gap.

**Comment by piotrmurach** [comment]
Released `tty-link` version `v0.2.0` that implements a brand new API that:
* supports Alacritty, Kitty, KDE's Konsole, and many more modern terminals. Please see the [Supported Terminals](https://github.com/piotrmurach/tty-link?tab=readme-ov-file#3-supported-terminals) section.
* allows hyperlink detection configuration with the `:hyperlink` keyword set to `:always`, `:auto` or `:never`.
* permits configuration with the `TTY_LINK_HYPERLINK` environment variable.
